packages/components/src/date-time/date.js

```diff
@@ -35,10 +36,24 @@ class DatePicker extends Component {
 		onChange( newDate.set( momentTime ).format( TIMEZONELESS_FORMAT ) );
 	}
 
+	/**
+	 * Create a Moment object from a date string. With no currentDate supplied, default to a Moment
+	 * object representing now. If a null value is passed, return a null value.
+	 *
+	 * @param {string|null} [currentDate] - Date representing the currently selected date or null to signify no selection.
+	 * @return {Moment|null} - Moment object for selected date or null.
```

Comment:
Similarly, there should be a newline separating the `@return` and `@param` tag segments.

https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions

```suggestion
	 * @return {?Moment} Moment object for selected date or null.
```

**Edit:** I see from the link above there's a different syntax for describing the optional parameter using the square bracket notation instead of the nullable type `?`. So I guess it's technically valid (preferred?) but not what's actually used through most of Gutenberg.